src/chainbase.cpp

```diff
@@ -172,6 +165,15 @@ namespace chainbase {
          _meta->flush();
    }
 
+   void database::_msync_database() {
+#ifdef _WIN32
+#warning Safe database dirty handling not implemented on WIN32
+#else
+         msync(_segment->get_address(), _segment->get_size(), MS_SYNC);
+         msync(_meta->get_address(), _meta->get_size(), MS_SYNC);
```

Comment:
Can we capture and log the return code of msync calls to possibly give us more information on failure.

Reply:
not a bad idea but we don't have fc::logger here what would you suggest?

Reply:
We use std::cerr to "log" mlock failure, so I guess just cerr.